core/templates/pages/exploration-editor-page/suggestion-modal-for-editor-view/exploration-editor-suggestion-modal.controller.spec.ts

```diff
@@ -0,0 +1,271 @@
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Unit tests for ExplorationEditorSuggestionModalController.
+ */
+
+// TODO(#7222): Remove the following block of unnnecessary imports once
+// the code corresponding to the spec is upgraded to Angular 8.
+import { UpgradedServices } from 'services/UpgradedServices';
+// ^^^ This block is to be removed.
+
+describe('Exploration Editor Suggestion Modal Controller', function() {
+  var $scope = null;
+  var $uibModalInstance = null;
+  var EditabilityService = null;
+  var SuggestionModalService = null;
+
+  var currentContent = 'Current Content';
+  var newContent = 'New Content';
+  var suggestionStatus = 'rejected';
+
+  beforeEach(angular.mock.module('oppia', function($provide) {
+    var ugs = new UpgradedServices();
+    for (let [key, value] of Object.entries(ugs.getUpgradedServices())) {
+      $provide.value(key, value);
+    }
+  }));
+
+  describe('when suggestion is already rejected', function() {
+    var suggestionIsHandled = true;
+    var suggestionIsValid = true;
+    var threadUibModalInstance = null;
+    var unsavedChangesExist = true;
+
+    beforeEach(angular.mock.inject(function($injector, $controller) {
+      var $rootScope = $injector.get('$rootScope');
+      EditabilityService = $injector.get('EditabilityService');
+      SuggestionModalService = $injector.get('SuggestionModalService');
+
+      $uibModalInstance = jasmine.createSpyObj(
+        '$uibModalInstance', ['close', 'dismiss']);
+
+      spyOn(EditabilityService, 'isEditable').and.returnValue(true);
+
+      $scope = $rootScope.$new();
+      $controller('ExplorationEditorSuggestionModalController', {
+        $scope: $scope,
+        $uibModalInstance: $uibModalInstance,
+        currentContent: currentContent,
+        newContent: newContent,
+        suggestionIsHandled: suggestionIsHandled,
+        suggestionIsValid: suggestionIsValid,
+        suggestionStatus: suggestionStatus,
+        threadUibModalInstance: threadUibModalInstance,
+        unsavedChangesExist: unsavedChangesExist,
+      });
+    }));
+
+    it('should init the variables', function() {
+      expect($scope.isNotHandled).toEqual(false);
+      expect($scope.canEdit).toBe(true);
+      expect($scope.commitMessage).toBe('');
+      expect($scope.reviewMessage).toBe('');
+      expect($scope.canReject).toBe(false);
+      expect($scope.canAccept).toBe(false);
+      expect($scope.currentContent).toBe(currentContent);
+      expect($scope.newContent).toBe(newContent);
+    });
+
+    it('should evaluate error message', function() {
+      expect($scope.errorMessage).toBe(
```

Comment:
Hm, I don't understand this test case. Shouldn't you be doing some action that results in an error then check if the errorMessage matches the expected error? Ditto for similar tests below.

Reply:
I've removed it and I'm expecting in another spec.